lib/fs.js

```diff
@@ -648,6 +654,14 @@ fs.readSync = function(fd, buffer, offset, length, position) {
     offset = 0;
   }
 
+  if (length === 0) {
```

Comment:
@thefourtheye Since the fd itself is passed best we can attempt to do is read directly from it. But as @feross mentioned, there are cross platform issues that prevents this being done correctly. I don't think that assuming the fd is alright is a problem in this case.